### DISCUSSION TITLE
Does serverless require a server?

### DISCUSSION THREAD

**Discussion by OnkelTem** [vercel/next.js#32449] [Help] [answered]
Hi!

First, I want to say thanks to the tutorial authors. The guide is really detailed and working.

However it seems that I didn't get one of the main points of Next.js: is it a front-end thing or a full-stack? I would think it's a front-end, but then why does `page/api` exist? It's definitely a server-side code, no? Then why it's called **serverless**? Why vercel tutorial is called that serverless also while it allows to execute server-code? If you're meaning some server, they why you never name it - is it expressjs or what? And why you don't say about using it along with some other CMS, like a headless one? Total mess here guys.

**Comment by leerob** [comment, marked as answer]
Thank you! Glad you found it helpful.

Next.js enables you to create full-stack applications if you'd like. You can run server-side code in API Routes, as well as data fetching functions like `getStaticProps` and `getServerSideProps`.

**Serverless really just means you don't have to manage or configure the infrastructure yourself.** There are still servers behind the scenes taking care of the hard work for you 😄

**Comment by OnkelTem** [comment]
> Serverless really just means you don't have to manage or configure the infrastructure yourself. 

Ah, I see now LOL.

But wow, that means that vercel - as a hosting platform in this case - allows us to run some server-side code for free? This is not common, I didn't expect that.. generosity :) What if one runs some stupid calculations?
